src/fields2cover/types/Path.cpp

```diff
@@ -114,6 +114,81 @@ std::string Path::serializePath(size_t digit_precision) const {
   return res;
 }
 
+/**
+ * @brief Discretize the swath sections of the path and return a new path
+ * @param step_size Discretization step in [m]
+ * @param swath_speed Robot swath speed in [m/s]
+ * @return New path with swath now discretized 
+*/
+Path Path::discretize_swath(double step_size, double swath_speed) const {
+    // Create new path
+    Path new_path;
+
+    // Loop through all the points in the path
+    for (size_t i = 0; i < this->size(); i++)
+    {
+        // Check if the current point is a SWATH point
+        if (this->states.at(i).type == f2c::types::PathSectionType::SWATH)
+        {
+            // We know that swaths are represented by only two point so the next point will
+            // be the end of the swath
+            // Save start and end swath points to variables
+            Point start_point = this->states.at(i).point;
+            Point end_point = this->states.at(i+1).point;
```

Comment:
if (i == this->size() - 1) then this->states.at(i+1) gets out of the boundaries

Reply:
Does this mean that sometimes the very last element in the path can be a single swath element? As far as I've seen swaths always have two point's a start and end when using the straight line swaths. If this is not the case please let me know what all the possible cases are for swaths in the Path objects.